### ISSUE TITLE
How to change the legend and setting modal of custom indicator 

### ISSUE THREAD

**Issue by p3777dev** [amcharts/amcharts5#1738]
I am using the custom indicator feature to create my custom indicators. But I got an issue that I am not able to change the legend and setting modal name of the custom indicator .

Note : I am creating the custom indicator by extending the class of the pre-Existing indicators.

I am able to change the legend and setting modal name if I create the indicator using the  `am5stock.Indicator` . By adding in the _afterNew() method.

https://codepen.io/prince-work/pen/JjgjBMZ

**Comment by martynasma** [comment]
> I am able to change the legend and setting modal name if I create the indicator using the am5stock.Indicator . By adding in the _afterNew() method.

I don't see any issues with this approach, if it works for you.

**Comment by p3777dev** [comment]
@martynasma  As you can see in the image the modal color changed but the legend didn't changed .
![image](https://github.com/user-attachments/assets/cd45d452-da35-46b4-a4fc-99b64376d9cb)

You can check the codepen too.

**Comment by martynasma** [comment]
Do you mean that there is no colored marker in the legend?

**Comment by workingbuddy10** [comment]
No @martynasma , You can see in the custom indicator legend is showing MA ( ideally it should have shown My Indicator) .

**Comment by workingbuddy10** [comment]
the custom indicator modal is showing correct info as My Indicator
Whereas the  custom indicator legend was showing the inherited class name only.

**Comment by martynasma** [comment]
@workingbuddy10 are you working with @p3777dev?

Can I assume this is @p3777dev reply?

**Comment by martynasma** [comment]
> No @martynasma , You can see in the custom indicator legend is showing MA ( ideally it should have shown My Indicator) .

Legend uses indicator's `shortName` by default. You can specify it just like `name`:

```JavaScript
this._setDefault("name", "My Indicator");
this._setDefault("shortName", "My Indicator");
```

**Comment by workingbuddy10** [comment]
This worked!

**Comment by workingbuddy10** [comment]
But this doesn't works in the case of Volume profile indicator. @martynasma 
https://codepen.io/Ansh-m-the-reactor/pen/xxvGXNO?editors=0010

**Comment by martynasma** [comment]
We'll fix that in the next version. Thanks.

**Comment by martynasma** [comment]
Fixed in [5.10.6](https://github.com/amcharts/amcharts5/releases/tag/5.10.6).

## [5.10.6] - 2024-10-11

### Added
- `hideDrawingGrips` setting (default: `false`) added to `StockChart`. If set to true, grips of drawings will be hidden when not in drawing mode.
- `toggleDrawing(enabled: boolean)` method added to `StockChart`. Toggles drawing mode on or off.
- `autoOpenSettings` setting (default: `true`) added to `Indicator`. If set to `false`, a settings modal will not pop up automatically when the indicator is added to chart via indicator control.

### Changed
- On `StockChart` labels and icons are no longer draggable to avoid interfering with chart drag/zoom.
- Arrow position slightly changed in "Line Arrow" drawing tool on `StockChart`. Grip at the end of the arrow removed.
- Arrow tip size will not increased to reflect the line thickness of a "Line Arrow".
- "Volume Profile" indicator had its full name hardcoded for legend. It will now use its `shortName` (which is "Volume Profile", too, by default but can be changed to anything else).
- `Modal` content will now have its contents aligned to `start`, rather than `left`, to properly handle RTL direction.

### Fixed
- "Line Arrow" drawing tool would not save its color when serialized.
- "Line Arrow" tip was not taking a color change.
- ES2025 examples involving `MapChart` were not correctly importing geodata.
- Using `clearDrawings()` was erroring if there was "Line Arrow" drawings added to `StockChart`.
- Rotated elements with HTML content were not being properly positioned in all cases.
- `scrollToChild` method of `Container` was not working properly.

[Full change log](https://github.com/amcharts/amcharts5/blob/master/packages/shared/CHANGELOG.md).

[Download options](https://www.amcharts.com/download/).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
